### ISSUE TITLE
`-mc-from` still creates snapshots of all past days

### ISSUE THREAD

**Issue by shawark** [instaclustr/supportCenter#38]
Run agent with `-mc-from` with today's date, I can see the tool try to create snapshots of all past days and later delete them.
```
./agent_darwin_amd64 -disable_known_hosts -pk ~/.ssh/id_rsa -config settings.yml -mc-from "2020-10-14T04:27:53+00:00" -l ubuntu

...
[2020-10-14T21:43:06+11:00]  INFO Block 1/23 01EKDTMQMM7EZC57ZMGJF4VAMV  2020-09-29 00:00:00 +0000 UTC .. 2020-09-29 18:00:00 +0000 UTC: will be skipped
[2020-10-14T21:43:06+11:00]  WARN Failed to drop snapshot block: Failed to remove resource '/data/ma/prometheus/snapshots/20201014T104302Z-162d968d3f71f8cb/01EKDTMQMM7EZC57ZMGJF4VAMV/' (SSH agent: Failed to run command 'rm -rf /data/ma/prometheus/snapshots/20201014T104302Z-162d968d3f71f8cb/01EKDTMQMM7EZC57ZMGJF4VAMV/' on '3.128.132.94'. (Process exited with status 1))
[2020-10-14T21:43:07+11:00]  INFO Block 2/23 01EKFRE7Q12FNTVKRJN29EJPFB  2020-09-29 18:00:00 +0000 UTC .. 2020-09-30 12:00:00 +0000 UTC: will be skipped
[2020-10-14T21:43:07+11:00]  WARN Failed to drop snapshot block: Failed to remove resource '/data/ma/prometheus/snapshots/20201014T104302Z-162d968d3f71f8cb/01EKFRE7Q12FNTVKRJN29EJPFB/' (SSH agent: Failed to run command 'rm -rf /data/ma/prometheus/snapshots/20201014T104302Z-162d968d3f71f8cb/01EKFRE7Q12FNTVKRJN29EJPFB/' on '3.128.132.94'. (Process exited with status 1))
[2020-10-14T21:43:08+11:00]  INFO NC 3.128.132.94: Collecting configuration files completed.
[2020-10-14T21:43:08+11:00]  INFO NC 3.128.132.94: Collecting log files...
[2020-10-14T21:43:08+11:00]  INFO Block 3/23 01EKHP7S3C8DVP70PZCYBKMP78  2020-09-30 12:00:00 +0000 UTC .. 2020-10-01 06:00:00 +0000 UTC: will be skipped
[2020-10-14T21:43:09+11:00]  INFO NC 3.131.19.74: Collecting configuration files completed.
[2020-10-14T21:43:09+11:00]  INFO NC 3.131.19.74: Collecting log files...
[2020-10-14T21:43:09+11:00]  WARN Failed to drop snapshot block: Failed to remove resource '/data/ma/prometheus/snapshots/20201014T104302Z-162d968d3f71f8cb/01EKHP7S3C8DVP70PZCYBKMP78/' (SSH agent: Failed to run command 'rm -rf /data/ma/prometheus/snapshots/20201014T104302Z-162d968d3f71f8cb/01EKHP7S3C8DVP70PZCYBKMP78/' on '3.128.132.94'. (Process exited with status 1))
[2020-10-14T21:43:09+11:00]  INFO Block 4/23 01EKKM1AZE4GA8D93KBN878SK3  2020-10-01 06:00:00 +0000 UTC .. 
...
```

Due to an issue in MetricsAppliance https://github.com/instaclustr/metricsAppliance/issues/4, these temporary snapshots cannot be deleted.

Not sure how Prometheus works, but first create snapshots of all past days and then delete create a lot of temporary files. With the combination issue https://github.com/instaclustr/metricsAppliance/issues/4, it can cause disk issue on remote server.

**Comment by imarsv** [comment]
The original Prometheus API do not have time span options. [(Prometheus snapshot)](https://prometheus.io/docs/prometheus/latest/querying/api/#snapshot)
That is why we create snapshot, than we remove unnecessary blocks based on [storage structure](https://prometheus.io/docs/prometheus/latest/storage/#local-storage)

[Feature request](https://github.com/instaclustr/supportCenter/issues/15)

**Comment by mrlika** [comment]
@shawark ,

The agent's user on Prometheus node has no permissions to delete snapshot created by Prometheus. It is not agent issue, it is configuration issue of Prometheus node.

Possible ways to solve it:
- Configure agent's user on the Prometheus node to have permissions to delete the snapshots
- Configure Prometheus to create snapshots that can be deleted by agent's user (not sure if Prometheus has this configuration)
- Run a cron job on Prometheus node that deletes old snapshots
- Install on Prometheus node a script the is able to delete Prometheus logs (i.e. with setuid bit set or similar). The agent will run the script to delete the files

**Comment by shawark** [comment]
Yeah agree the root cause is Prometheus permission but `supportcenter` is what could actually fill up the disk.

> Configure agent's user on the Prometheus node to have permissions to delete the snapshots

The user need to be either the user running Prometheus (which is a service user not SSH-able) or `root` (not all customer setup allow root access) because the tmp files created is `rwx-rx-rx` or `rx-r-r`

> Configure Prometheus to create snapshots that can be deleted by agent's user (not sure if Prometheus has this configuration)

Yeah, that is something the customer side need to be aware of (so we need to document in the README). However since https://github.com/instaclustr/metricsAppliance/issues/4 is still open, I suspect that is not out of the box.

> Run a cron job on Prometheus node that deletes old snapshots
> Install on Prometheus node a script the is able to delete Prometheus logs (i.e. with setuid bit set or similar). The agent will run the script to delete the files

Again, we need to document if the customer wishes to use `supportcenter`.

So again, it is a prometheus issue but it is also Supportcenter issue (because it is supportcenter that actually filled up the disk). We need to document how to work around the issue. Otherwise, this becomes the blocker to use supportcenter.

**Comment by jordanbraiuka** [comment]
Given that this is ultimately due to a side effect of the prometheus snapshot method, and the permissions that the snapshot is created with, its probably adequate to log at the end of the support centre script that we were unable to clear snapshots from the prometheus server, and list the files that will need to be manually cleaned up.
